src/NUnitFramework/tests/Constraints/IndexerConstraintTests.cs

```diff
@@ -0,0 +1,177 @@
+        }
+
+        [Test]
+        public void DoesNotMatchMissingIndexer()
+        {
+            var expectedErrorMessage = $"  Expected: Default indexer accepting arguments < 42.0d >{NL}  But was:  <NUnit.Framework.Constraints.IndexerConstraintTests+IndexerTester>{NL}";
+            
+            var tester = new IndexerTester();
+            
+            var ex = Assert.Throws<AssertionException>(() => Assert.That(tester, Has.Item(42d)));
+            Assert.That(ex.Message, Is.EqualTo(expectedErrorMessage));
+        }
+
+        [Test]
+        public void DoesNotMatchMissingIndexerEquality()
+        {
+            var expectedErrorMessage = $"  Expected string length 14 but was 13. Strings differ at index 0.{NL}  Expected: \"Second indexer\"{NL}  But was:  \"Third indexer\"{NL}  -----------^{NL}";
+            
+            var tester = new IndexerTester();
+
+            var ex = Assert.Throws<AssertionException>(() => Assert.That(tester, Has.Item(4, 2).EqualTo("Second indexer")));
+            Assert.That(ex.Message, Is.EqualTo(expectedErrorMessage));
+        }
+
+        [Test]
+        public void CanMatchGenericIndexer()
+        {
+            var tester = new GenericIndexerTester<int>(100);
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(42).EqualTo(100));
+        }
+
+        [Test]
+        public void CanMatchNamedIndexer()
+        {
+            var tester = new NamedIndexTester();
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(42).EqualTo("A Named Int Indexer"));
+            Assert.That(tester, Has.Item(42, 43).EqualTo("A Named Int Int Indexer"));
+        }
+
+        [Test]
+        public void CanMatchShadowIndexer()
+        {
+            var tester = new DerivedClassWithoutIndexer();
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(string.Empty).EqualTo("Second indexer from shadow"));
+            Assert.That(tester, Has.Item(1, 2).EqualTo("Third indexer from shadow"));
+            Assert.That(tester, Has.Item(41).EqualTo("Still calculating"));
+        }
+
+        [Test]
+        public void CanMatchShadowNamedIndexer()
+        {
+            var tester = new DerivedClassWithoutNamedIndexer();
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(42).EqualTo("A Named Int Indexer from shadow"));
+            Assert.That(tester, Has.Item(42, 43).EqualTo("A Named Int Int Indexer"));
+        }
+
+        private class IndexerTester
+        {
+            public int DummyParam { get; set; }
+
+            public string this[int x] => x == 42 ? "Answer to the Ultimate Question of Life, the Universe, and Everything" : "Still calculating";
+
+            public string this[string x] => "Second indexer";
+
+            protected string this[int x, int y] => "Third indexer";
+        }
+
+        private class ClassWithShadowIndexer : IndexerTester
```

Comment:
Nitpick: The `new` member in the derived class is usually called the *shadowing* or *hiding* indexer and the base member is the *shadowed* or *hidden* member.

Also, I only recently learned that the C# lingo is primarily 'hide,' not 'shadow' which is terminology I guess I carried over from my VB days a decade ago. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/new-modifier

Reply:
Oh, yes :man_facepalming: I even read the error message:
```
because it hides inherited member
```
and didn't twig :rofl: